classes/issue/IssueDAO.inc.php

```diff
@@ -586,12 +586,12 @@ function getByPubIdType($pubIdType, $contextId = null, $pubIdSettingName = null,
 				LEFT JOIN issue_settings ist ON (i.issue_id = ist.issue_id)
 				'. ($pubIdSettingName != null?' LEFT JOIN issue_settings iss ON (i.issue_id = iss.issue_id AND iss.setting_name = ?)':'') .'
 			WHERE
-				ist.setting_name = ? AND ist.setting_value IS NOT NULL
+				i.published = 1 AND ist.setting_name = ? AND ist.setting_value IS NOT NULL
```

Comment:
Wow, yes, you are right, it makes sense! I will do that as well, but i will leave the `i.published = 1` above, ok?